integrations/anthropic/src/haystack_integrations/components/generators/anthropic/generator.py

```diff
@@ -52,7 +55,9 @@ class AnthropicGenerator:
 
     def __init__(
         self,
-        api_key: Secret = Secret.from_env_var("ANTHROPIC_API_KEY"),  # noqa: B008
+        api_key: Optional[Secret] = Secret.from_env_var("ANTHROPIC_API_KEY"),  # noqa: B008
+        region: Optional[str] = None,
+        project_id: Optional[str] = None,
```

Comment:
All parameters are kept optional, allowing users to select which ones to provide based on the targeted model. However, without an explicit option for users to choose between Vertex and Anthropic, it may lead to confusion. WDYT?